code/modules/reagents/chemistry/holder.dm

```diff
@@ -66,6 +65,12 @@
 	var/list/datum/reagent/addiction_list
 	/// various flags, see code\__DEFINES\reagents.dm
 	var/flags
+	///list of reactions currently on going
+	var/list/datum/equilibrium/reaction_list = new/list()
+	///cached list of reagents typepaths (not object references)
+	var/list/datum/reagent/previous_reagent_list = new/list()
```

Comment:
```suggestion
	///list of reactions currently on going
	var/list/datum/equilibrium/reaction_list = list()
	///cached list of reagents typepaths (not object references)
	var/list/datum/reagent/previous_reagent_list = list()
```
Same as the other, consider making them lazy-lists if they're often empty.

Reply:
I'm not sure how to do lazy lists, I can't find an example in code, other than the ones that leave the list null. Could you maybe point me to an example?